### PR TITLE
Makes Explosion and Singulo Defer Based on CPU

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This is from TG.

Basically uses BYOND's built-in tick duration to detct when a tick is about to be driven into overtime, but instead of being applied to the scheduler and process loops, it's being applied to two of the most CPU intensive things in the game; the singulo and explosions.

When an explosion takes place, for each thing it performs `ex_act` on, it will check the current tick usage---if it's about to go into overtime, it'll sleep for the the current tick rate so it doesn't cause as much *perceptible* lag to players.

You can create size 40-60-80 explosion with no perceptible lag to players with this.

Due to the nature of deferring, explosion will progress over a time frame; the more lag they would cause, the longer the time frame--as such, explosions now fan outward in a spiral, so they behave more like...well, explosions.

related
- Removes Cael code that used the now unused `trange` proc; couldn't be bothered to fix up unused horrible code

Requesting @Tastyfish test this out, as well.

### CODE REVIEW

File: code/game/objects/objs.dm [hunk 1]
```diff
@@ -53,6 +53,7 @@
 /obj/item/proc/is_used_on(obj/O, mob/user)
 
 /obj/proc/process()
+	set waitfor = 0
```
Comment (Fox-McCloud) [inline code/game/objects/objs.dm]: This is so sleeps don't lock up the scheduler---though, I suppose I could just a `spawn` to `eat` for the singulo, but this seems like the better way to handle it.

**Comment by monster860** [comment]
Epic.

**Comment by IK3I** [comment]
A size 80 explosion is going to be something you'll hear before you see, which will give you just enough warning to start running before you are brutally gibbed. 10/10 would explode again.

**Comment by Keilden** [comment]
Now we need a jump to the ground command so we can escape fireballs!

**Comment by Bigtimetripod** [comment]
☝ There is a rest command for it. If ye get lucky and delay doesnt eff ye over, it is possible to dodge one i think. 

Not sure if it was changed or not, but having throwing active with a empty hand (catch mode) you could for a second catch what wizard is "throwing" at you.

**Comment by TheDZD** [comment]
Not really that feasibly to outrun these unless an admin makes something like a 40-60-80 explosion, and you're towards the tail end of the outermost radius, these are still pretty damn fast.

**Comment by Fox-McCloud** [comment]
@TheDZD 

Cool spessmen, however, may be able to actually turn to look at the explosion.

**Pull request merged**